File: test/python/bench_scaled/map_lookup.py
"""Scaled mirror of test/jacl/bench_scaled/map_lookup.jacl — 3M lookups = 493733."""
def run():
    m = {}
    for i in range(200):
        m[i] = i * 7
    acc = 0
    for j in range(3000000):
        k = (j * 73 + 11) % 200
        acc = (acc + m[k]) % 1000003
    return acc
if __name__ == "__main__":
    print(run())

File: test/lua/bench_scaled/map_lookup.lua
-- Scaled mirror of test/jacl/bench_scaled/map_lookup.jacl — 3M lookups = 493733.
local function run()
  local m = {}
  for i = 0, 199 do m[i] = i * 7 end
  local acc = 0
  for j = 0, 2999999 do
    local k = (j * 73 + 11) % 200
    acc = (acc + m[k]) % 1000003
  end
  return acc
end
return { run = run }
